internal/usage/usage.go

```diff
@@ -0,0 +1,114 @@
+package usage
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"runtime"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/grafana/synthetic-monitoring-agent/internal/feature"
+	sm "github.com/grafana/synthetic-monitoring-agent/pkg/pb/synthetic_monitoring"
+)
+
+// Reporter represents a way of communicating reports to different backend systems.
+type Reporter interface {
+	ReportProbe(ctx context.Context, probe sm.Probe) error
+}
+
+// report represents a specific usage event that will be sent to https://stats.grafana.com to be processed and stored.
+// Each attribute represents a column in a BigQuery table that can be easily searched.
+// Adding new attributes to report will not automatically update the table, and instead needs to be handled in https://github.com/grafana/usage-stats
+type report struct {
+	CreatedAt    string `json:"createdAt"`
+	OS           string `json:"os"`
+	Arch         string `json:"arch"`
+	Report       string `json:"report"`
+	Version      string `json:"version"`
+	Public       bool   `json:"public"`
+	Features     string `json:"features"`
+	UsageStatsId string `json:"usageStatsId"`
+	TenantID     int64  `json:"tenantId"`
+	ProbeID      int64  `json:"probeId"`
+}
+
+// HTTPReporter represents
+type HTTPReporter struct {
+	endpoint string
+	features feature.Collection
+	client   *http.Client
+}
+
+const (
+	// UsageReportApplication aligns with the usage-stats service endpoint defined
+	// in github.com/grafana/usage-stats for synthetic monitoring agents
+	UsageStatsApplication = "synthetic-monitoring-agent-usage-report"
+	// Base Endpoint for usage stats
+	DefaultUsageStatsEndpoint = "https://stats.grafana.com"
+)
+
+func NewHTTPReporter(endpoint string, features feature.Collection) Reporter {
+	if endpoint == "" {
+		endpoint = DefaultUsageStatsEndpoint
+	}
+	return &HTTPReporter{
+		endpoint: endpoint,
+		features: features,
```

Comment:
In relation to this [comment](https://github.com/grafana/synthetic-monitoring-agent/pull/1366#discussion_r2180859033).
Yes, I think passing the _features_ to the reporter constructor is "not ideal" as it's not a field of the reporter, it's a field that represents certain characteristics of the probe.

Based on that, should we remove this from the reporter itself and accept it as part of the `ReportProbe()` method?
Notice that, if I'm not wrong, the _features_ are also a field of the `Updater`, therefore we can pass them from `loop()` [here](https://github.com/grafana/synthetic-monitoring-agent/blob/1a2dbdc358cbde6fe7b6660de1bc1f583a13f205/internal/checks/checks.go#L391) as:
```
	err = c.usageReporter.ReportProbe(ctx, result.Probe, c.features)
```